### ISSUE TITLE
Certain component, helper, and modifier imports fail to resolve with v2.0.0

### ISSUE THREAD

**Issue by kiosion** [embroider-build/embroider#1293]
Building with `staticHelpers`/`staticModifiers`/`staticComponents` set to `false` results in `@embroider/compat` failing to resolve `*.hbs.js` imports for certain template-only components, throwing a build error:
```
Build Error (PackageRunner) in components/maybe-in-element.js

Module not found: Error: Can't resolve '../node_modules/ember-maybe-in-element/components/maybe-in-element' in '$TMPDIR/embroider/40cd52/components/maybe-in-element.js'
```

Building with the same options set to `true` succeeds, but results in certain package components, helpers, and modifiers failing to resolve at runtime instead:
```
Uncaught Error: Attempted to resolve `ensure-safe-component`, which was expected to be a helper, but nothing was found.
    at resolveHelper (opcode-compiler.js:276:1)
    at encodeOp (opcode-compiler.js:2607:1)
    at pushOp (opcode-compiler.js:2521:1)
    at opcode-compiler.js:545:1
    at Compilers.compile (opcode-compiler.js:520:1)
    at expr (opcode-compiler.js:727:1)
    at Curry (opcode-compiler.js:972:1)
    at opcode-compiler.js:559:1
    at Compilers.compile (opcode-compiler.js:520:1)
    at expr (opcode-compiler.js:727:1)
```

I've had some success changing [compat/src/synthesize-template-only-components.ts#L10](https://github.com/embroider-build/embroider/blob/main/packages/compat/src/synthesize-template-only-components.ts#L10) to `['.hbs', '.hbs.js']` and looping over that in `crawl()`, although this only seems to fix the first issue

**Comment by ef4** [comment]
Thanks for a thorough report. It's enough to help me understand at least the first part of the problem.

The thing that changed here in 2.0 is that if an addon has a custom AST transform, its templates will now get converted from `.hbs` to `.hbs.js` in the addon compatibility step. (The reason for this is that AST transforms can now, in general, emit Javascript alongside their changes to the contents of the template.) Before they were still `.hbs`, just with modified contents.

After that change I adjusted the template-colocation case (so a .js will find its corresponding .hbs.js) but I probably forgot to adjust the template-only case (and I guess we had a gap in test coverage).

I'm not sure about the second part of your problem, I don't know why ensure-safe-component wouldn't resolve. That can become a separate issue if it's still a problem after dealing with this first one.

**Comment by kiosion** [comment]
Gotcha, should I submit a PR to deal with the first half? Making that change to check for `.hbs.js` as well does seem to fix the build errors

**Comment by ef4** [comment]
Fixed and released in 2.0.1.
